src/middlewared/middlewared/pytest/unit/plugins/pool/test_dataset_recordsize_choices.py

```diff
@@ -0,0 +1,66 @@
+import pytest
+from unittest.mock import Mock
+
+from middlewared.plugins.pool_.dataset_recordsize import PoolDatasetService
+
+OBJ = PoolDatasetService(Mock())
+MAPPING = OBJ.MAPPING
+EXPECTED_RESULT = {}
+for raw_val, human_readable in MAPPING.items():
+    if human_readable == '512B':
+        EXPECTED_RESULT[raw_val] = set((human_readable,))
+    elif human_readable == '1K':
+        EXPECTED_RESULT[raw_val] = set(('512B', human_readable))
+    elif human_readable == '2K':
+        EXPECTED_RESULT[raw_val] = set(('512B', '1K', human_readable))
+    elif human_readable == '4K':
+        EXPECTED_RESULT[raw_val] = set(('512B', '1K', '2K', human_readable))
+    elif human_readable == '8K':
+        EXPECTED_RESULT[raw_val] = set(('512B', '1K', '2K', '4K', human_readable))
+    elif human_readable == '16K':
+        EXPECTED_RESULT[raw_val] = set(('512B', '1K', '2K', '4K', '8K', human_readable))
+    elif human_readable == '32K':
+        EXPECTED_RESULT[raw_val] = set(('512B', '1K', '2K', '4K', '8K', '16K', human_readable))
+    elif human_readable == '64K':
+        EXPECTED_RESULT[raw_val] = set(('512B', '1K', '2K', '4K', '8K', '16K', '32K', human_readable))
+    elif human_readable == '128K':
+        EXPECTED_RESULT[raw_val] = set(('512B', '1K', '2K', '4K', '8K', '16K', '32K', '64K', human_readable))
+    elif human_readable == '256K':
+        EXPECTED_RESULT[raw_val] = set(('512B', '1K', '2K', '4K', '8K', '16K', '32K', '64K', '128K', human_readable))
+    elif human_readable == '512K':
```

Comment:
This is excessive. A function that maps `/sys/module/zfs/parameters/zfs_max_recordsize` contents to the list of the possible values would be smaller than this implementation + test 😃 One middle value (e.g. `512K` should be sufficient), it will make the entire function 100% covered.

Reply:
Worrying about how "small" something is in code doesn't really make any logical sense. It accomplishes what it sets out to do just fine. Whether or not it uses 100 lines or 10 lines is irrelevant imo 😄